src/main/java/com/iexec/worker/pubsub/SubscriptionService.java

```diff
@@ -75,16 +75,25 @@ public SubscriptionService(CoreConfigurationService coreConfigurationService,
 
     @PostConstruct
     private void run() {
-        this.connectStomp();
+        this.restartStomp();
     }
 
-    private void connectStomp() {
+    private void restartStomp() {
+        log.info("Starting STOMP");
         WebSocketClient webSocketClient = new StandardWebSocketClient();
-        stompClient = new WebSocketStompClient(webSocketClient);
-        stompClient.setMessageConverter(new MappingJackson2MessageConverter());
-        stompClient.setTaskScheduler(new ConcurrentTaskScheduler());
-        stompClient.connect(url, this);
-        log.info("Connect STOMP [isRunning: {}]", stompClient.isRunning());
+        this.stompClient = new WebSocketStompClient(webSocketClient);
+        this.stompClient.setMessageConverter(new MappingJackson2MessageConverter());
+        this.stompClient.setTaskScheduler(new ConcurrentTaskScheduler());
+        this.stompClient.connect(url, this);
+        this.reSubscribeToTopics();
+        log.info("Started STOMP");
+    }
+
+    private void reSubscribeToTopics() {
+        for (String chainTaskId : chainTaskIdToSubscription.keySet()) {
```

Comment:
Can we iterate on a copy of the set? It is not a good idea to manipulate the set while iterating on it.